### DISCUSSION TITLE
404 on entry to any route other than root

### DISCUSSION THREAD

**Discussion by Neuroforge** [vercel/community#942] [Help] [answered]
Hello,

I have a React app that is deployed to Vercel via the CLI. It is not yet in production so i am using the default link provided post deployment.

It seems the routes are somehow protected. I am unable to go directly to my-app.vercel.app/a/sign-in (404 error) but if i go to my-app.vercel.app (root url) it correctly redirects to my-app.vercel.app/a/sign-in (via the router). I want to redirect my users via a verification email to a specific page on my app but running into 404 error.

Any attempt to go anywhere except the root url causes a 404 error.

I was looking at https://vercel.com/docs/project-configuration#legacy/routes but this does not seem to be the correct place to solve my issue.

When using localhost, the routing works as expected. So it seems Vercel is doing something odd that i do not understand.

Any advice would be appreciated on this issue.

This seems to be the most similar issue on the Github community. https://github.com/vercel/community/discussions/799

**Comment by Neuroforge** [comment, marked as answer]
Hello. After some more searching, i found this solution that appears to have worked. I hope it can help someone else!

https://stackoverflow.com/questions/64815012/why-does-react-router-not-works-at-vercel

**Reply by amyegan** [reply]
Thanks for coming back to share this answer! I'm glad you found a solution that works for your project 😌

**Reply by bencipher** [reply]
Your solution helped me in 2024, thank you.
